File: yaml file conversion/uvlm_readmodel.py
import re
import numpy as np
from matplotlib import pyplot as plt
def uvlm_readmodel():
	model = []
	fid1 = open("simulationinput_aero.txt","r")
	fid = -1
    range_4to6 = np.array([4,5,6])
		if (fid1 != -1):
			fid = fid1
		if (fid >= 0): 
			for i in range (1,4):
				tline = fid.readline()		
			tline = fid.readline().strip()
			str_n = list(map(str,tline.split("")))
			string = str_n
            count = len(str_n)
			if (count>1):
				str_n = list(str(str_n))
				string = dict(str_n[1][1])
				string = string[1]			
			model["strSimName"] = string
			for i in range (1,4):
				tline = fid.readline()			
			tline = fid.readline().strip()
            tline = re.sub('d','e',tline)
			model["simulationsettings"] = list(map(float,tline.split("")))
			for i in range (1,4):
				tline = fid.readline()                
			tline = fid.readline().strip()
            tline = re.sub('d','e',tline[9:-1])
			model["windData"] = list(map(float,tline.split(" ")))
			model["density"]  = model["windData"][1]
			model["winddir"]  = model["windData"][range_4to6]
			model["vinf"]     = model["windData"][2]*model["winddir"]
			plt.close(fid)
		
	fid3 = open("simulationinput_fsi.txt","r")
	if (fid3 >= 0):
		fid = fid3
		for i in range (1,4):
			tline = fid.readline()		
		tline = fid.readline().strip()
		str_n = list(map(str,tline.split("")))
        string = str_n
        count = len(str_n)
		if (count > 1):
			str_n = list(str(str_n))
            string = dict(str_n[1][1])
            string = string[1]		
		model["strSimName"] = string
		for i in range (1,4):
			tline = fid.readline()
		tline = fid.readline().strip()
        tline = re.sub('d','e',tline)
		model["simulationsettings"] = list(map(float,tline.split("")))
		for i in range (1,4):
			tline = fid.readline()
		tline = fid.readline().strip()
        tline = re.sub('d','e',tline[9:-1])
		model["windData"] = list(map(float,tline.split("")))
		model["density"]  = model["windData"][1]
		model["winddir"]  = model["windData"][range_4to6]
		model["vinf"]     = model["windData"][2]*model["winddir"]       
		plt.close(fid3)

	fid1 = open(model["strSimName"] + '_uvlm_models.dres',"r")
	fid2 = open(model["strSimName"] + '_uvlm_models.txt',"r")
	fid = -1
    range_1to3 = np.array([1,2,3])
		if (fid1 != -1):
			fid = fid1
		elseif (fid2 != -1):
			fid = fid2
	if (fid >= 0): 
		tline = readline(fid)
		model["nsurfaces"] = sscanf(tline,'%i')
		i3 = 0
		for i in range (1,model["nsurfaces"]+1):
			tline = fid.readline().strip()
			arrline = list(map(int,tline.split("")))
			model["surfaces"][i]["nnodes"] = arrline[1]
			model["surfaces"][i]["nnodes"] = arrline[4]
			for j in range (1,model["surfaces"][i]["nelements"] + 1):
				tline = fid.readline().strip()
				model["surfaces"][i]["connectivity"][j,:] = list(map(int,tline.split("")))
			for j in range (1,model["surfaces"][i]["nnodes"]):				
                model["surfaces"][i]["nodes"][j]["indices_q"] = [3*(j-1)+(range_1to3)*(j-1)+3] + i3           
			i3 = i3 + 3*model["surfaces"][i]["nnodes"]		
	plt.close(fid)
	return
    
    

File: yaml_and_converter_matlab_test_codes/Pre_post_processing/DeSiO-Aero/uvlm_readmodel.m
function model = uvlm_readmodel()
% =================================================================================================================
model = [];
fid1 = fopen(['simulationinput_aero.txt'],'r');
fid2 = fopen(['simulationinput.txt'],'r');
fid = -1;
    if fid1 ~= -1
        fid = fid1;
    elseif fid2 ~= -1
        fid = fid2;
    end
    if fid >= 0 
        for i = 1:3
            tline = fgets(fid);
        end
        tline = fgetl(fid);
        [str_n,count] = sscanf(tline,'%s%c');
        str = str_n;
        if count>1
            str_n = textscan(str_n,'%s');
            str = cellstr(str_n{1}(1));
            str = str{1};
        end
        model.strSimName = str;
        for i = 1:3
            tline = fgets(fid);
        end
        tline = fgetl(fid); tline = strrep(tline,'d','e');
        model.simulationsettings = sscanf(tline,'%f',3);
        for i = 1:3
            tline = fgets(fid);
        end
        tline = fgetl(fid); tline = strrep(tline(10:end),'d','e');
        model.windData = sscanf(tline,'%f',6);
        model.density  = model.windData(1);
        model.winddir  = model.windData(4:6);
        model.vinf     = model.windData(2)*model.winddir;
        fclose(fid);
    end
    
fid3 = fopen(['simulationinput_fsi.txt'],'r');
if fid3 >= 0
    fid = fid3;
    for i = 1:3
        tline = fgets(fid);
    end
    tline = fgetl(fid);
    [str_n,count] = sscanf(tline,'%s%c');
    str = str_n;
    if count>1
        str_n = textscan(str_n,'%s');
        str = cellstr(str_n{1}(1));
        str = str{1};
    end
    model.strSimName = str;
    for i = 1:3
        tline = fgets(fid);
    end
    tline = fgetl(fid); tline = strrep(tline,'d','e');
    model.simulationsettings = sscanf(tline,'%f',4);
    for i = 1:3
        tline = fgets(fid);
    end
    tline = fgetl(fid); tline = strrep(tline(10:end),'d','e');
    model.windData = sscanf(tline,'%f',6);
    model.density  = model.windData(1);
    model.winddir  = model.windData(4:6);
    model.vinf     = model.windData(2)*model.winddir;        
    fclose(fid3);
end

fid1 = fopen([model.strSimName '_uvlm_models.dres'],'r');
fid2 = fopen([model.strSimName '_uvlm_models.txt'],'r');
fid = -1;
    if fid1 ~= -1
        fid = fid1;
    elseif fid2 ~= -1
        fid = fid2;
    end
if fid >= 0 
    tline = fgetl(fid);
    model.nsurfaces = sscanf(tline,'%i');
    i3 = 0;
    for i = 1:model.nsurfaces
        tline = fgetl(fid);
        arrline = sscanf(tline,'%i',5);
        model.surfaces(i).nnodes     = arrline(1);
        model.surfaces(i).nelements = arrline(4);
        for j = 1:model.surfaces(i).nelements
            tline   = fgetl(fid);
            model.surfaces(i).connectivity(j,:) = sscanf(tline,'%i',4);
        end
        for j = 1:model.surfaces(i).nnodes
            model.surfaces(i).nodes(j).indices_q = [3*(j-1)+1:3*(j-1)+3] + i3;
        end
        i3 = i3 + 3*model.surfaces(i).nnodes;
    end
fclose(fid);
end

% =================================================================================================================
return